pkg/resources/database.go

```diff
@@ -147,13 +196,105 @@ func ReadDatabase(d *schema.ResourceData, meta interface{}) error {
 	}
 
 	err = d.Set("data_retention_time_in_days", i)
-	return err
+	if err != nil {
+		return err
+	}
+
+	stmt = snowflake.Replication(name).Show()
+	row = snowflake.QueryRow(db, stmt)
+	replication, err := snowflake.ScanReplication(row)
+
+	if err != nil {
+		if err == sql.ErrNoRows {
+			// If not found, no replication is enabled
+			d.Set("replication_accounts", make([]string, 0))
+			d.Set("replication_enable_failover", false)
+			d.Set("replication_is_primary", false)
+			return nil
+		}
+		return err
+	}
+
+	err = d.Set("replication_accounts", strings.Split(replication.ReplAccounts.String, ", "))
+	if err != nil {
+		return err
+	}
+
+	err = d.Set("replication_failover_accounts", strings.Split(replication.FailoverAccounts.String, ", "))
+	if err != nil {
+		return err
+	}
+
+	return d.Set("replication_is_primary", replication.IsPrimary.Bool)
 }
 
 func UpdateDatabase(d *schema.ResourceData, meta interface{}) error {
+	db := meta.(*sql.DB)
+	database := d.Get("name").(string)
+
+	// Change the replication details first - this is a special case and won't work using the generic method
+	x := func(resource, replType string, toggleGrant func(db *sql.DB, database, account, replType, toggleType string) error) error {
+		o, n := d.GetChange(resource)
```

Comment:
Can you use more descriptive names for these variables?

Reply:
Well, this is how it is in other resources, but sure, changed to old_ver/new_ver.